azure_costs_exporter/scrape.py

```diff
@@ -53,17 +53,27 @@ def convert_json_df(data):
 
     return df
 
-def extract_metrics_from_df(df, counter):
+def extract_metrics_from_df(df, metric_name, enrollment, registry):
     """
     Fill a counter object with the data from the data frame
     """
     groups = df.groupby(base_columns).sum()
-    for name, value in groups.iterrows():
-        meta = dict(zip(base_columns, name))
-        counter.record(value.ExtendedCost, **meta)
 
+    class AzureCollector(object):
+        def collect(self):
+            c = CounterMetricFamily(metric_name,
+                                    "Costs billed to Azure Enterprise Agreement {}".format(enrollment),
+                                    labels=base_columns)
 
-def query_metrics(enrollment, token, metric_name, month=None):
+            for name, value in groups.iterrows():
+                c.add_metric(name, value.ExtendedCost)
+
+            yield c
+
+    registry.register(AzureCollector())
```

Comment:
The key to making work what Michael proposed here is to use the default `REGISTRY` rather than a custom `CollectorRegistry`.
 (https://github.com/prometheus/client_python/blob/ce5542bd8be2944a1898e9ac3d6e112662153ea4/prometheus_client/__init__.py#L12)